compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -300,6 +312,9 @@ pub fn from_fn_attrs(cx: &CodegenCx<'ll, 'tcx>, llfn: &'ll Value, instance: ty::
     // Without this, ThinLTO won't inline Rust functions into Clang generated
     // functions (because Clang annotates functions this way too).
     apply_target_cpu_attr(cx, llfn);
+    // tune-cpu is only conveyed through the attribute for our purpose.
+    // The target doesn't care; the subtarget reads our attribute.
+    apply_tune_cpu_attr(cx, llfn);
```

Comment:
Does tune-cpu have any of the same issues as noted for target-cpu above, affecting cross-language LTO?

Reply:
Since it's the only way this gets conveyed in Clang, kind of?

Reply:
More directly, does the mismatched presence or absence of this attribute interfere with optimizations like inlining?

Reply:
No.